### PR TITLE
Add draft go-selinux project proposal

### LINKED ISSUE #27: Consideration: moving selinux library out of runc into a new project
There's been a discussion in the OCI TDC about potentially moving the selinux library out of runc and into a new project (say something like opencontainers/libselinux)

I'm creating this issue as a placeholder while the OCI TDC finishes debating whether to move forward or not. If we move forward, we will consult the TOB and begin a formal project proposal.

cc: @mrunalp

### PR DESCRIPTION
**Pull request by caniszczyk**
Closes #27 

Signed-off-by: Chris Aniszczyk <caniszczyk@gmail.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

File: proposals/selinux.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+### Versioning / Roadmap
+We will probably minimize the releases of this project.
+
+## Frequently Asked Questions (FAQ)
+Q: Does this change the OCI Charter or Scope Table?
+A: No.  Nothing in this proposal is intended to amend the OCI Charter (https://www.opencontainers.org/about/governance) or OCI Scope Table (https://www.opencontainers.org/about/oci-scope-table).
+
+Q: Why move this out of the runc project?
+A: TODO
```
Comment (mrunalp) [inline proposals/selinux.md]: To be able to reuse this in different container projects as well as have dedicated maintainers for the SELinux library.

**Comment by mrunalp** [comment]
Added one comment to address TODO. Otherwise LGTM. @caniszczyk Thanks for putting this together!

File: proposals/selinux.md (hunk 1 above)
Comment (crosbymichael) [inline proposals/selinux.md]: dedicated maintainers for stability

**Comment by crosbymichael** [comment]
LGTM

**Comment by caniszczyk** [comment]
@philips as chair of the @opencontainers/tob, any comments before we bring this up for a vote?

**Comment by philips** [comment]
FAQ items I want to see:

- Why is versioning this package with runc insufficient today? What issues have been encountered?
- How will the release management of go-selinux work and get vendored into runc?
- What is blocking sub-maintainers from being a concept in runc itself?
- Who are the other target users of go-selinux?

**Comment by rhatdan** [comment]
I have been personally asked for these go bindings outside of runc/libcontainer over the years.

**Comment by philips** [comment]
@rhatdan but why a new git repo and release process? Is it to raise awareness? Have a separate release process because runc release too slow? Essentially what is the motivation.

**Comment by rhatdan** [comment]
I was asked if I was  interested, and I said sure.  But I can't say I had a huge motivation.  

That being said...
Getting more exposure and others to use it would probably lead to completing lots of features that are missing from the libcontainer/selinux bindings.  There are lots of bindings in libselinux that do not have native bindings yet.  Getting other projects to use SELinux bindings would also lead to potential improvements in the bindings.  I am not sure if we sucked them into RKT when we do the SELinux work on it, or just referenced them from libcontainer.

Being able to get fixes into go-selinux without having to wait for runc approval would also be nice.

**Comment by runcom** [comment]
runc can also stick to a particular version w/o the hassle of people changing the selinux package causing bugs when upgrading (it happened)

**Comment by runcom** [comment]
> Why is versioning this package with runc insufficient today? What issues have been encountered?

no versioning of selinux in runc - for instance, we fixed something in selinux in runc because CRI-O needed it but at the same time we broke docker which was relying on it. Having fixed versions for selinux wouldn't have led to this issue since docker could have stuck to a previous version and carefully test the new version w/o pulling new changes as part of a libcontainer library bump.

> How will the release management of go-selinux work and get vendored into runc?
> What is blocking sub-maintainers from being a concept in runc itself?

nothing I guess. To me it's clearer to have a dedicated repo for just the selinux library (as what we already have today with go-digest in OCI)

> Who are the other target users of go-selinux?

docker, CRI-O, kubernetes(?), any other project out there requiring a dedicated selinux library. It'll be easy to add features in a dedicated repo also.

**Comment by rhatdan** [comment]
RKT also I believe.

**Comment by philips** [comment]
I don't really care either way but I trust @crosbymichael @rhatdan @runcom so:

LGTM

**Comment by philips** [comment]
@caniszczyk I can propose this to the TOC. What are the next steps? Merge and propose via email?

**Comment by caniszczyk** [comment]
@philips yep, you would have to do a @opencontainers/tob vote (see go-digest as an example: https://groups.google.com/a/opencontainers.org/forum/#!topic/tob/kplbAW5N9dc)

**Comment by philips** [comment]
done

File: README.md [hunk 2]
```diff
@@ -22,6 +22,7 @@ https://groups.google.com/a/opencontainers.org/forum/#!forum/tob (tob@opencontai
 
 * [Digest](https://github.com/opencontainers/tob/blob/master/proposals/digest.md)
 * [Image Format Spec](https://github.com/opencontainers/tob/tree/master/proposals/image-format)
+* [Selinux](https://github.com/opencontainers/tob/blob/master/proposals/selinux.md)
```
Comment (wking) [inline README.md]: I'm fine with all lower case, but where we uppercase anything I think we should use "SELinux".

File: proposals/selinux.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+https://github.com/runcom/go-selinux
+
+### Initial Maintainers
+Initial maintainers of the go-selinux project would be:
+
+* Antonio Murdaca <runcom@redhat.com> (@runcom)
+* Daniel J Walsh <dwalsh@redhat.com> (@rhatdan)
+* Mrunal Patel <mpatel@redhat.com> (@mrunalp)
+* TODO
```
Comment (wking) [inline proposals/selinux.md]: Fill this in before the TOB vote completes?

File: proposals/selinux.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+### Code of Conduct
+This project would incorporate (by reference) the OCI Code of Conduct.
+
+### Governance and Releases
+This project would incorporate the Governance and Releases processes from the OCI project template: https://github.com/opencontainers/project-template.
+
+### Project Communications
+Both of the proposed projects would continue to use existing channels in use by the OCI developer community for communication including:
```
Comment (wking) [inline proposals/selinux.md]: Both?  Copy/paste from somewhere else?  I think there's only one project here, unless you mean runC is not leaving.  Either way, I think this line could be more clear (e.g. listing projects by name).

File: proposals/selinux.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+* The dev@opencontainers.org email list
+* The weekly OCI developer community conference call
+* The #OpenContainers IRC channel
+
+### Versioning / Roadmap
+We will probably minimize the releases of this project.
+
+## Frequently Asked Questions (FAQ)
+Q: Does this change the OCI Charter or Scope Table?
+A: No.  Nothing in this proposal is intended to amend the OCI Charter (https://www.opencontainers.org/about/governance) or OCI Scope Table (https://www.opencontainers.org/about/oci-scope-table).
```
Comment (AkihiroSuda) [inline proposals/selinux.md]: nit: markdown formatting is not pretty

image

**Comment by caniszczyk** [comment]
We have enough votes to approve this before the vote closes out but I want to do a final call for maintainers on this project, right now I have three:

* Antonio Murdaca <runcom@redhat.com> (@runcom)
* Daniel J Walsh <dwalsh@redhat.com> (@rhatdan)
* Mrunal Patel <mpatel@redhat.com> (@mrunalp)

cc: @opencontainers/runc-maintainers

**Comment by caniszczyk** [comment]
Also to note we have 8/9 +1s from the @opencontainers/tob on this so we have well cleared the minimum bar of votes needed for this project to happen.

**Comment by rhatdan** [comment]
I have invited Stephen Smalley to be a maintainer, (SELinux upstream maintainer) but that should not slow this down.

**Comment by caniszczyk** [comment]
@rhatdan what's his GitHub id btw?

**Comment by rhatdan** [comment]
https://github.com/stephensmalley PTAL

**Comment by caniszczyk** [comment]
We have enough votes here, @philips will call the vote later tonight on the mailing list.

Thanks @opencontainers/tob and everyone for participating.

**Pull request merged**